plugins/catalog-backend/src/service/util.ts

```diff
@@ -148,75 +148,21 @@ export function decodeCursor(encodedCursor: string) {
 // sure that all adopters have re-stitched their entities so that the new
 // targetRef field is present on them, and that they have stopped consuming
 // the now-removed old field
-// TODO(jhaals): Remove this in April 2022
-export function expandLegacyCompoundRelationRefsInResponse(
-  entities: Entity[],
-): void {
-  for (const entity of entities) {
-    if (entity.relations) {
-      for (const relation of entity.relations as any) {
-        if (!relation.targetRef && relation.target) {
-          // This is the case where an old-form entity, not yet stitched with
-          // the updated code, was in the database
-          relation.targetRef = stringifyEntityRef(relation.target);
-        } else if (!relation.target && relation.targetRef) {
-          // This is the case where a new-form entity, stitched with the
-          // updated code, was in the database but we still want to produce
-          // the old data shape as well for compatibility reasons
-          relation.target = parseEntityRef(relation.targetRef);
-        }
+// TODO(patriko): Remove this in catalog 2.0
+export function expandLegacyCompoundRelationsInEntity(entity: Entity): Entity {
+  if (entity.relations) {
+    for (const relation of entity.relations as any) {
+      if (!relation.targetRef && relation.target) {
+        // This is the case where an old-form entity, not yet stitched with
+        // the updated code, was in the database
+        relation.targetRef = stringifyEntityRef(relation.target);
+      } else if (!relation.target && relation.targetRef) {
+        // This is the case where a new-form entity, stitched with the
+        // updated code, was in the database but we still want to produce
+        // the old data shape as well for compatibility reasons
+        relation.target = parseEntityRef(relation.targetRef);
```

Comment:
Do you think that the sheer amount of `parseEntityRef` going on in one single linear job (when returning huge numbers of highly connected entities) could be a problem in and of itself, causing a large part of the blockage?

Reply:
Hmm, maybe, let's leave that for now though